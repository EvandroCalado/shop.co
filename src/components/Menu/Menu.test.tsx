import '@testing-library/jest-dom';
import { fireEvent, render, screen } from '@testing-library/react';
import { useState } from 'react';
import { Menu } from '.';

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
}));

describe('<Menu />', () => {
  it('should render with default values', () => {
    const mockSetShow = jest.fn();
    useState.mockReturnValue([false, mockSetShow]);

    render(<Menu />);

    expect(screen.getByTestId('menu')).toBeInTheDocument();
    expect(screen.getByTestId('menu-container')).toBeInTheDocument();
    expect(screen.getByTestId('menu-container')).toHaveClass('invisible');
    expect(screen.getByTestId('close')).toBeInTheDocument();

    expect(screen.getByTestId('menu-container')).toHaveClass(
      'invisible opacity-0',
    );
  });

  it('should render menu container visible', () => {
    const mockSetShow = jest.fn();
    useState.mockReturnValue([true, mockSetShow]);

    render(<Menu />);

    const menuButton = screen.getByTestId('menu');

    fireEvent.click(menuButton);

    expect(screen.getByTestId('menu-container')).toHaveClass(
      'visible opacity-100',
    );
  });

  it('should render menu container invisible', () => {
    const mockSetShow = jest.fn();
    useState.mockReturnValue([false, mockSetShow]);

    render(<Menu />);

    expect(screen.getByTestId('menu-container')).toHaveClass(
      'invisible opacity-0',
    );

    const closeButton = screen.getByTestId('close');

    fireEvent.click(closeButton);
  });

  it('should render a snapshot', () => {
    const { container } = render(<Menu />);

    expect(container).toMatchSnapshot();
  });
});
