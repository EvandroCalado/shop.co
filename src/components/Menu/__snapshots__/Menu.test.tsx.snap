// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> should render a snapshot 1`] = `
<div>
  <div
    class="md:hidden"
  >
    <button
      data-testid="menu"
    >
      <svg
        class="lucide lucide-menu "
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <line
          x1="4"
          x2="20"
          y1="12"
          y2="12"
        />
        <line
          x1="4"
          x2="20"
          y1="6"
          y2="6"
        />
        <line
          x1="4"
          x2="20"
          y1="18"
          y2="18"
        />
      </svg>
    </button>
    <div
      class="invisible opacity-0 absolute inset-0 z-10 m-auto flex h-[95%] w-[95%] flex-col items-center justify-center gap-4 bg-[#f0f0f0] capitalize transition-all duration-300"
      data-testid="menu-container"
    >
      <button
        class="absolute left-3 top-4 sm:left-[7px]"
        data-testid="close"
      >
        <svg
          class="lucide lucide-x text-red-600"
          fill="none"
          height="24"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 6 6 18"
          />
          <path
            d="m6 6 12 12"
          />
        </svg>
      </button>
      <a
        class="cursor-pointer duration-150 hover:text-zinc-600"
        href="/shop"
      >
        shop
      </a>
      <a
        class="cursor-pointer duration-150 hover:text-zinc-600"
        href="/on-sale"
      >
        on sale
      </a>
      <a
        class="cursor-pointer duration-150 hover:text-zinc-600"
        href="/new-arrivals"
      >
        new arrivals
      </a>
      <a
        class="cursor-pointer duration-150 hover:text-zinc-600"
        href="/brands"
      >
        brands
      </a>
    </div>
  </div>
</div>
`;
